### DISCUSSION TITLE
Why did OpenVINO switch from Doxyrest to Breathe?

### DISCUSSION THREAD

**Discussion by hwhsu1231** [openvinotoolkit/openvino#21773] [Q&A] [answered]
Hello, OpenVINO Development Team.

Recently, I've been researching the bridging usage between Sphinx and Doxygen. In this process, I found several solutions like Doxyrest, Breathe, Exhale, and Doxysphinx.

Currently, I'm primarily collecting and analyzing projects that have actually adopted these solutions. Then, I discovered that OpenVINO originally used Doxyrest, but switched to Breathe after merging this PR:

https://github.com/openvinotoolkit/openvino/pull/21239

I wonder what factors OpenVINO considered in deciding to discontinue Doxyrest and switch to Breathe?

**Comment by kblaszczak-intel** [comment, marked as answer]
Hi there,
I'll be happy to answer. We don't get many questions regarding documentation ;)
The basic reason was build simplicity and reliability. We used Doxygen for the entire documentation before and Sphinx was implemented hastily, without proper adjustments. Doxyrest must have seemed the quickest solution back then but later proved a bit problematic in our setup (for example, introducing unwanted characters in some conditions). Instead of fixing the issues, we decided to switch.
Breathe seems better integrated with Sphinx and is documented well enough to make the switch fairly painless. Maintenance looks easier, as it has fewer elements and dependencies, and control over the content looks more promising.
